app/src/main/java/com/lambdasoup/quickfit/alarm/AlarmReceiver.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016-2019 Juliane Lehmann <jl@lambdasoup.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.lambdasoup.quickfit.alarm
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.media.AudioAttributes
+import android.net.Uri
+import android.os.Build
+import android.provider.Settings
+import androidx.preference.PreferenceManager
+import com.lambdasoup.quickfit.Constants
+import com.lambdasoup.quickfit.R
+import com.lambdasoup.quickfit.persist.QuickFitContentProvider
+import com.lambdasoup.quickfit.util.WakefulIntents
+
+/**
+ * Receives broadcasts for PendingIntents scheduled via [android.app.AlarmManager]. Uses the contained info to immediately show the
+ * notification, and kicks off the necessary bookkeeping I/O work to (a) persist the notification state, so that it can be restored
+ * after boot and (b) register the next alarm.
+ */
+class AlarmReceiver : BroadcastReceiver() {
+    override fun onReceive(context: Context, intent: Intent) {
+        if (intent.action != ACTION_NOTIFY) {
```

Comment:
do you expect other actions?

Reply:
Not really - but spurious intent delivery on BroadcastReceivers is a thing, so I just always do the filtering.